### ISSUE TITLE
✅ 🐞 Vimeo private videos not loading in WP - add private hash to embed URL to fix

### ISSUE THREAD

**Issue by savduvimeo** [elementor/elementor#16741]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

Private videos added via Video widget in WordPress are not working. Player / embed URLs on private videos created after 9/14 require a private hash in addition to the video ID. The current implementation is stripping the private hash, causing these videos not to load. 

For example, for a private video, here is the URL being pasted in Elementor/WP: https://vimeo.com/640039734/e8cea1d27a. The player/embed URL should be https://player.vimeo.com/video/640039734?h=e8cea1d27a. However, Elementor currently strips the hash and links to https://player.vimeo.com/video/640039734, which doesn't work because it doesn't include the hash param. Calling the [Vimeo embed endpoint](https://developer.vimeo.com/api/oembed/videos) will supply the player URL directly, including the hash.

![Screen Shot 2021-10-28 at 2 33 19 PM](https://user-images.githubusercontent.com/86245548/139314993-c918b2ac-b276-451d-a423-9e54575a0b01.png)

## Steps to reproduce

Add private Vimeo video to WP page.
![Screen Shot 2021-10-28 at 2 25 49 PM](https://user-images.githubusercontent.com/86245548/139313819-49bf464c-8aa3-40ca-bf57-b29180ad5bed.png)

## Isolating the problem
- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
<!-- For bug reports, let us know about your system environment: Copy and paste the system info report from Elementor => System info, and paste it here or in http://pastebin.com/ -->
```
</details>

**Comment by caiobleggi** [comment]
Any news on that? Almost an year and still cant use private videos...

**Comment by Wecoboss** [comment]
I also support this bug fix request. I can't embed private or unlisted videos hosted on Vimeo inside the playlist widget. It would be really nice to fix this bug. 

Many thanks in advance

**Comment by jordane2086** [comment]
I am experiencing this same issue with a Private Vimeo video and using the "Video Playlist" element. However, it works fine on the Media Carousel and regular video elements. Any idea why this is happening and how to fix it ASAP?

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor v3.23.0! 🥳 

Now Vimeo videos that have privacy settings enabled should load correctly!

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
